homeassistant/components/hausbus/__init__.py

```diff
@@ -0,0 +1,124 @@
+"""Integration for all haus-bus.de modules."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.exceptions import ConfigEntryNotReady, HomeAssistantError
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.device_registry import DeviceEntry
+
+from .const import DOMAIN
+from .gateway import HausbusGateway
+
+CONFIG_SCHEMA = cv.empty_config_schema(DOMAIN)
+
+PLATFORMS: list[Platform] = [
+    Platform.COVER,
+]
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class HausbusConfig:
+    """Class for Hausbus ConfigEntry."""
+
+    gateway: HausbusGateway
+
+
+HausbusConfigEntry = ConfigEntry[HausbusConfig]
```

Comment:
If we're going to only store the gateway in the config entry, might as well skip the dataclass